### ISSUE TITLE
Examples need better explanation

### ISSUE THREAD

**Issue by JohnTheCoolingFan** [makspll/bevy_mod_scripting#53]
I'm reading examples to understand how exactly can I use this crate for my project. So far it wasn't very easy to understand, I had to go back and forth: exampels, script assets, API docs. And also I don't understand some comments completely, like [this one](https://github.com/makspll/bevy_mod_scripting/blob/b666e56fcce9885a7e7f3e9f3103bdf206e15d69/examples/lua/event_recipients.rs#L27-L29). It looks like it's out of place.

This is why I think exampels need improvements to the explanation comments and more such comments.

**Comment by makspll** [comment]
I completely agree, the examples started out relatively sane but the library underwent some significant changes since its conception so some comments went astray

**Comment by makspll** [comment]
The examples were condensed now and I think combined with the better docs this should be addressed :D
